### PR TITLE
Wrote about conditional constant initialization

### PR DESCRIPTION
**Pull request by vandadnp**
@dakeshi would you please review this PR? Thank you 🙏🏻☺️

### CODE REVIEW

File: chapters/constants.md [hunk 1]
```diff
@@ -88,10 +86,27 @@ let up: Direction = .Up
 
 ## [Conditional Initialization](#conditional-initialization)
 
-TODO: write about how we can create a constant that doesn't have an initial value and how we can set this value conditionally with if statements later before use
+A constant can be conditionally initialized using the `if` statement. This means that the constant will **not** have an initial value and instead its initial value will be set using a conditional. Let's have a look at an example:
```
Comment (dakeshi) [inline chapters/constants.md]: I think that we can add the description about a [ternary conditional operator](https://developer.apple.com/library/content/documentation/Swift/Conceptual/Swift_Programming_Language/BasicOperators.html#//apple_ref/doc/uid/TP40014097-CH6-ID60)
which can make the same result with more shorthand form.

If you agree with this, the above sentences can be updated with:

> A constant can be conditionally initialized. This means that the constant will **not** have an initial value and instead its initial value will be set using `if` statement or `ternary conditional operator`. Let's have a look at an example:

And we can add a ternary conditional operator concepts, examples, and shortcoming in the end of this section.

Thoughts?

File: chapters/constants.md [hunk 2]
```diff
@@ -88,10 +86,27 @@ let up: Direction = .Up
...
+if name == "Foo"{
+ isFoo = true
+} else {
+ isFoo = false
+}
+
+isFoo //true
+```
 
+In this example, the constant `name` contains a string, in this case, `"Foo"`. Right after this, we are defining a constant called `isFoo` that does **not** have an initial value. However, right after the definition of this constant, we use the `if` statement to set the value of this constant, called `isFoo`.
```
Comment (dakeshi) [inline chapters/constants.md]: > However, right after the definition of this constant, we use the `if` statement to set the value of this constant, called `isFoo`.

I think that declaration would be more natural in the context and `isFoo` is redundant.
So, I suggest the following sentence:

> However, right after the declaration of this constant, 
we set the value of this constant using `if` statement.

File: chapters/constants.md [hunk 3]
```diff
@@ -88,10 +86,27 @@ let up: Direction = .Up
...
 
+Note two things about this arrangement:
 
+1. You have to somehow set the value of your uninitialized constants before using them
```
Comment (dakeshi) [inline chapters/constants.md]: how was this?

> It is not allowed to use the constant without a value.

File: chapters/constants.md [hunk 4]
```diff
@@ -1,14 +1,14 @@
 # Constants
 
-A constant is a storage place for *a value*, with a name and a type which can be implicitly or explicitly specified.
+A constant is a storage place for _a value_, with a name and a type which can be implicitly or explicitly specified.
 
 A constant's value cannot be changed after it has been created and named.
 
 ## [Syntax](#syntax)
 
 The `let` statement allows you to create a constant. What follows after `let` is the name of the constant that has to follow the [CamelCase naming convention](https://en.wikipedia.org/wiki/CamelCase).
```
Comment (vandadnp) [inline chapters/constants.md]: I agree, thank you.

File: chapters/constants.md [hunk 5]
```diff
@@ -1,14 +1,14 @@
...
 
-Here is an example of a constant called *nameOfYourConstant* whose value is *xyz*. This constant has no explicit data type. Once you assign a value to a constant, Swift will figure out the data type automatically.
+Here is an example of a constant called _nameOfYourConstant_ whose value is _xyz_. This constant has no explicit data type. Once you assign a value to a constant, Swift will figure out the data type automatically.
```
Comment (vandadnp) [inline chapters/constants.md]: GitBook editor did this. It is now fixed. Great catch!

File: chapters/constants.md (hunk 1 above)
Comment (vandadnp) [inline chapters/constants.md]: This is great, I will use this for the new material. Thanks.

**Comment by vandadnp** [comment]
I'm going to have to delete this PR since the GitBook editor has really messed up the content. The latest version has fixed this. I will send another PR. Thank you @dakeshi for looking at this.

**Pull request closed without merging**